packages/protocol/ledger/ledger.go

```diff
@@ -293,13 +335,16 @@ func (l *Ledger) propagateAcceptanceToIncludedTransactions(txID utxo.Transaction
 
 // propagateConfirmedConflictToIncludedTransactions propagates confirmations to the included future cone of the given
 // Transaction.
-func (l *Ledger) propagatedRejectionToTransactions(txID utxo.TransactionID) {
-	l.Storage.CachedTransactionMetadata(txID).Consume(func(txMetadata *TransactionMetadata) {
+func (l *Ledger) propagatedRejectionToTransactions(conflict *conflictdag.Conflict[utxo.TransactionID, utxo.OutputID]) {
+	l.Storage.CachedTransactionMetadata(conflict.ID()).Consume(func(txMetadata *TransactionMetadata) {
```

Comment:
We should lock the outer one too no?

Reply:
It should not be needed: it is pretty much the same as `propagateAcceptanceToIncludedTransactions` above. We only lock this TX for writing when we set its state in the `triggerRejectedEventLocked` . 
But we have to make sure again that the booking with rejection does not race with a fork with rejection.